src/App.tsx

```diff
@@ -1,40 +1,65 @@
-import React, { ReactElement } from 'react';
+import React, { ReactElement, useEffect } from 'react';
 import {
   BrowserRouter as Router,
   Route,
   Switch,
   RouteComponentProps,
 } from 'react-router-dom';
 import 'hds-core';
+import querystring, { ParsedUrlQuery } from 'querystring';
+import {
+  Tokens,
+  setTokens,
+  getTokens,
+  convertParamsToTokens,
+  isValidAuthParams,
+  AuthContext,
+} from './auth/auth-context';
 import Main from './components/main/Main';
 import NavigationAndFooterWrapper from './components/navigation-and-footer-wrapper/NavigationAndFooterWrapper';
 import './App.scss';
 import TargetPage from './target/page/TargetPage';
 
 export default function App(): JSX.Element {
+  const existingTokens: Tokens | undefined = getTokens();
+
+  const parameterTokens: ParsedUrlQuery = querystring.parse(
```

Comment:
Could we apply term authParams here? Would better fit with the functions `isValidAuthParams` and `converParamsToTokens` :)

Reply:
Renamed